app/models/provider.rb

```diff
@@ -98,6 +98,10 @@ def current_accredited_courses
     accredited_courses.includes(:provider).where(provider: { recruitment_cycle: })
   end
 
+  scope :case_insensitive_search, lambda { |provider_code|
+    where('lower(provider_code) = ?', provider_code.to_s.downcase).first!
```

Comment:
Since you've called `.first!` on this scope, wondering what your thoughts are on calling `.first!` on this one too: https://github.com/DFE-Digital/publish-teacher-training/blob/96296adca32cac1658b697c341a1120be86251a1/app/models/course.rb#L201-L203

Instead of doing it in the `apply` method? Not sure if that will affect anything, so just a question.